ray_on_golem/server/services/ray.py

```diff
@@ -142,28 +147,33 @@ async def request_nodes(
         return created_node_ids
 
     async def _create_node(self, node_id: NodeId) -> None:
-        activity, ip, ssh_proxy_command = await self._golem_service.create_activity(
-            node_config=self._provider_config.node_config,
-            public_ssh_key=self._ssh_public_key,
-            ssh_user=self._ssh_user,
-            ssh_private_key_path=self._ssh_private_key_path,
-            total_budget=self._provider_config.total_budget,
-            payment_network=self._provider_config.payment_network,
-            subnet_tag=self._provider_config.subnet_tag,
-            add_state_log=partial(self._add_node_state_log, node_id),
-        )
+        try:
```

Comment:
why are we wrapping all the code here in a try block instead of just wrapping the `create_activity` call?

Reply:
Well, in this case, `_create_node()` is scheduled as a task, and is run in the background. There is no point in the code to await that task, hence it needs to wrap itself with `try/catch` instead of relying on the caller.